pact/Pact/Core/SPV.hs

```diff
@@ -1,22 +1,37 @@
 module Pact.Core.SPV
- ( ContProof(..)
- , SPVSupport(..)
- , noSPVSupport)where
-
-import Data.Text(Text)
-import Data.ByteString(ByteString)
+  ( -- * Types
+    ContProof(..)
+  , SPVSupport(..)
+    -- * Support
+  , noSPVSupport
+  ) where
+
+import Data.Aeson hiding (Object)
+import qualified Pact.JSON.Encode as J
+import Data.Text (Text)
+import Data.Text.Encoding
+import Data.ByteString (ByteString)
 import GHC.Generics
 import Control.DeepSeq
+import Control.Lens (Wrapped)
 
 import Pact.Core.DefPacts.Types
 import Pact.Core.PactValue
 
 newtype ContProof = ContProof { _contProof :: ByteString }
   deriving (Eq, Ord, Show, Generic)
 
+instance Wrapped ContProof
 
 instance NFData ContProof
 
+instance J.Encode ContProof where
```

Comment:
Why do we need an `Encode` instance for `ContProof` ?

Reply:
Most likely to encode it in outputs

Reply:
It wouldn't go into outputs, but it would tx data that we serialize.